From ec7c9556b15519348ec7af471cbce54d276e577a Mon Sep 17 00:00:00 2001
From: Zygmunt Krynicki <me@zygoon.pl>
Date: Mon, 6 Mar 2017 20:02:02 +0100
Subject: [PATCH 3/7] cmd: add directive for shellcheck and follow source:

shellcheck 0.4.4 needs a directive to find the common.sh script that is
sourced by all the seccomp unit tests. We also need to use the '-x'
command line option to have shellcheck follow such declarations.

Signed-off-by: Zygmunt Krynicki <me@zygoon.pl>
---
 cmd/Makefile.am                                                         | 2 +-
 cmd/snap-confine/tests/test_bad_seccomp_filter_args                     | 1 +
 cmd/snap-confine/tests/test_bad_seccomp_filter_args_clone               | 1 +
 cmd/snap-confine/tests/test_bad_seccomp_filter_args_null                | 1 +
 cmd/snap-confine/tests/test_bad_seccomp_filter_args_prctl               | 1 +
 cmd/snap-confine/tests/test_bad_seccomp_filter_args_prio                | 1 +
 cmd/snap-confine/tests/test_bad_seccomp_filter_args_quotactl            | 1 +
 cmd/snap-confine/tests/test_bad_seccomp_filter_args_socket              | 1 +
 cmd/snap-confine/tests/test_bad_seccomp_filter_args_termios             | 1 +
 cmd/snap-confine/tests/test_bad_seccomp_filter_length                   | 1 +
 cmd/snap-confine/tests/test_bad_seccomp_filter_missing_trailing_newline | 1 +
 cmd/snap-confine/tests/test_complain                                    | 1 +
 cmd/snap-confine/tests/test_complain_missed                             | 1 +
 cmd/snap-confine/tests/test_noprofile                                   | 1 +
 cmd/snap-confine/tests/test_restrictions                                | 1 +
 cmd/snap-confine/tests/test_restrictions_working                        | 1 +
 cmd/snap-confine/tests/test_restrictions_working_args                   | 1 +
 cmd/snap-confine/tests/test_restrictions_working_args_clone             | 1 +
 cmd/snap-confine/tests/test_restrictions_working_args_prctl             | 1 +
 cmd/snap-confine/tests/test_restrictions_working_args_prio              | 1 +
 cmd/snap-confine/tests/test_restrictions_working_args_quotactl          | 1 +
 cmd/snap-confine/tests/test_restrictions_working_args_socket            | 1 +
 cmd/snap-confine/tests/test_restrictions_working_args_termios           | 1 +
 cmd/snap-confine/tests/test_unrestricted                                | 1 +
 cmd/snap-confine/tests/test_unrestricted_missed                         | 1 +
 cmd/snap-confine/tests/test_whitelist                                   | 1 +
 26 files changed, 26 insertions(+), 1 deletion(-)

diff --git a/cmd/Makefile.am b/cmd/Makefile.am
index 4352451..574886c 100644
--- a/cmd/Makefile.am
+++ b/cmd/Makefile.am
@@ -28,7 +28,7 @@ check-syntax-c:
 .PHONY: check-syntax-sh
 check-syntax-sh: $(filter-out %.log %.trs,$(wildcard $(srcdir)/snap-confine/tests/test_*)) snap-confine/tests/common.sh
 if HAVE_SHELLCHECK
-	$(HAVE_SHELLCHECK) --format=gcc $^
+	$(HAVE_SHELLCHECK) -x --format=gcc $^
 else
 	echo "shellcheck is not installed, skipping (non-critical) syntax test"
 endif
diff --git a/cmd/snap-confine/tests/test_bad_seccomp_filter_args b/cmd/snap-confine/tests/test_bad_seccomp_filter_args
index 900dcb7..f3f4104 100755
--- a/cmd/snap-confine/tests/test_bad_seccomp_filter_args
+++ b/cmd/snap-confine/tests/test_bad_seccomp_filter_args
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 get_common_syscalls >"$TMP"/tmpl
diff --git a/cmd/snap-confine/tests/test_bad_seccomp_filter_args_clone b/cmd/snap-confine/tests/test_bad_seccomp_filter_args_clone
index 7ac29ec..0eae33f 100755
--- a/cmd/snap-confine/tests/test_bad_seccomp_filter_args_clone
+++ b/cmd/snap-confine/tests/test_bad_seccomp_filter_args_clone
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 get_common_syscalls >"$TMP"/tmpl
diff --git a/cmd/snap-confine/tests/test_bad_seccomp_filter_args_null b/cmd/snap-confine/tests/test_bad_seccomp_filter_args_null
index 259127e..17e9c04 100755
--- a/cmd/snap-confine/tests/test_bad_seccomp_filter_args_null
+++ b/cmd/snap-confine/tests/test_bad_seccomp_filter_args_null
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 get_common_syscalls >"$TMP"/tmpl
diff --git a/cmd/snap-confine/tests/test_bad_seccomp_filter_args_prctl b/cmd/snap-confine/tests/test_bad_seccomp_filter_args_prctl
index e05bdde..d364ce7 100755
--- a/cmd/snap-confine/tests/test_bad_seccomp_filter_args_prctl
+++ b/cmd/snap-confine/tests/test_bad_seccomp_filter_args_prctl
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 get_common_syscalls >"$TMP"/tmpl
diff --git a/cmd/snap-confine/tests/test_bad_seccomp_filter_args_prio b/cmd/snap-confine/tests/test_bad_seccomp_filter_args_prio
index 6872b02..5ae6b48 100755
--- a/cmd/snap-confine/tests/test_bad_seccomp_filter_args_prio
+++ b/cmd/snap-confine/tests/test_bad_seccomp_filter_args_prio
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 get_common_syscalls >"$TMP"/tmpl
diff --git a/cmd/snap-confine/tests/test_bad_seccomp_filter_args_quotactl b/cmd/snap-confine/tests/test_bad_seccomp_filter_args_quotactl
index 07b7329..91499f5 100755
--- a/cmd/snap-confine/tests/test_bad_seccomp_filter_args_quotactl
+++ b/cmd/snap-confine/tests/test_bad_seccomp_filter_args_quotactl
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 get_common_syscalls >"$TMP"/tmpl
diff --git a/cmd/snap-confine/tests/test_bad_seccomp_filter_args_socket b/cmd/snap-confine/tests/test_bad_seccomp_filter_args_socket
index 2cf477a..58675bc 100755
--- a/cmd/snap-confine/tests/test_bad_seccomp_filter_args_socket
+++ b/cmd/snap-confine/tests/test_bad_seccomp_filter_args_socket
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 get_common_syscalls >"$TMP"/tmpl
diff --git a/cmd/snap-confine/tests/test_bad_seccomp_filter_args_termios b/cmd/snap-confine/tests/test_bad_seccomp_filter_args_termios
index d6d7d91..4493e16 100755
--- a/cmd/snap-confine/tests/test_bad_seccomp_filter_args_termios
+++ b/cmd/snap-confine/tests/test_bad_seccomp_filter_args_termios
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 get_common_syscalls >"$TMP"/tmpl
diff --git a/cmd/snap-confine/tests/test_bad_seccomp_filter_length b/cmd/snap-confine/tests/test_bad_seccomp_filter_length
index d2a1548..0d06005 100755
--- a/cmd/snap-confine/tests/test_bad_seccomp_filter_length
+++ b/cmd/snap-confine/tests/test_bad_seccomp_filter_length
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 printf "Test seccomp filter (bad - too long)"
diff --git a/cmd/snap-confine/tests/test_bad_seccomp_filter_missing_trailing_newline b/cmd/snap-confine/tests/test_bad_seccomp_filter_missing_trailing_newline
index ea6bcf2..9c77844 100755
--- a/cmd/snap-confine/tests/test_bad_seccomp_filter_missing_trailing_newline
+++ b/cmd/snap-confine/tests/test_bad_seccomp_filter_missing_trailing_newline
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 printf "Test seccomp filter (bad - no trailing newline)"
diff --git a/cmd/snap-confine/tests/test_complain b/cmd/snap-confine/tests/test_complain
index 4b7878a..b58de26 100755
--- a/cmd/snap-confine/tests/test_complain
+++ b/cmd/snap-confine/tests/test_complain
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 cat >"$TMP/snap.name.app" <<EOF
diff --git a/cmd/snap-confine/tests/test_complain_missed b/cmd/snap-confine/tests/test_complain_missed
index 9267f57..d6b0e19 100755
--- a/cmd/snap-confine/tests/test_complain_missed
+++ b/cmd/snap-confine/tests/test_complain_missed
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 cat >"$TMP/snap.name.app" <<EOF
diff --git a/cmd/snap-confine/tests/test_noprofile b/cmd/snap-confine/tests/test_noprofile
index 5ac2dcb..ef98e9e 100755
--- a/cmd/snap-confine/tests/test_noprofile
+++ b/cmd/snap-confine/tests/test_noprofile
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 printf "Test that a non-existing profile causes the launcher to not start"
diff --git a/cmd/snap-confine/tests/test_restrictions b/cmd/snap-confine/tests/test_restrictions
index 8794b65..364ad9b 100755
--- a/cmd/snap-confine/tests/test_restrictions
+++ b/cmd/snap-confine/tests/test_restrictions
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 cat >"$TMP/snap.name.app" <<EOF
diff --git a/cmd/snap-confine/tests/test_restrictions_working b/cmd/snap-confine/tests/test_restrictions_working
index d4df229..b01e109 100755
--- a/cmd/snap-confine/tests/test_restrictions_working
+++ b/cmd/snap-confine/tests/test_restrictions_working
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 get_common_syscalls >"$TMP/snap.name.app"
diff --git a/cmd/snap-confine/tests/test_restrictions_working_args b/cmd/snap-confine/tests/test_restrictions_working_args
index 7db3c63..09a300c 100755
--- a/cmd/snap-confine/tests/test_restrictions_working_args
+++ b/cmd/snap-confine/tests/test_restrictions_working_args
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 get_common_syscalls >"$TMP"/tmpl
diff --git a/cmd/snap-confine/tests/test_restrictions_working_args_clone b/cmd/snap-confine/tests/test_restrictions_working_args_clone
index 2c43db3..9700d3e 100755
--- a/cmd/snap-confine/tests/test_restrictions_working_args_clone
+++ b/cmd/snap-confine/tests/test_restrictions_working_args_clone
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 get_common_syscalls >"$TMP"/tmpl
diff --git a/cmd/snap-confine/tests/test_restrictions_working_args_prctl b/cmd/snap-confine/tests/test_restrictions_working_args_prctl
index 7c2f53c..6717726 100755
--- a/cmd/snap-confine/tests/test_restrictions_working_args_prctl
+++ b/cmd/snap-confine/tests/test_restrictions_working_args_prctl
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 get_common_syscalls >"$TMP"/tmpl
diff --git a/cmd/snap-confine/tests/test_restrictions_working_args_prio b/cmd/snap-confine/tests/test_restrictions_working_args_prio
index dfbbc22..40d31e2 100755
--- a/cmd/snap-confine/tests/test_restrictions_working_args_prio
+++ b/cmd/snap-confine/tests/test_restrictions_working_args_prio
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 get_common_syscalls >"$TMP"/tmpl
diff --git a/cmd/snap-confine/tests/test_restrictions_working_args_quotactl b/cmd/snap-confine/tests/test_restrictions_working_args_quotactl
index f9c1376..ab7056c 100755
--- a/cmd/snap-confine/tests/test_restrictions_working_args_quotactl
+++ b/cmd/snap-confine/tests/test_restrictions_working_args_quotactl
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 get_common_syscalls >"$TMP"/tmpl
diff --git a/cmd/snap-confine/tests/test_restrictions_working_args_socket b/cmd/snap-confine/tests/test_restrictions_working_args_socket
index dfbe5ec..a50c632 100755
--- a/cmd/snap-confine/tests/test_restrictions_working_args_socket
+++ b/cmd/snap-confine/tests/test_restrictions_working_args_socket
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 get_common_syscalls >"$TMP"/tmpl
diff --git a/cmd/snap-confine/tests/test_restrictions_working_args_termios b/cmd/snap-confine/tests/test_restrictions_working_args_termios
index 26ef73b..0c2a42a 100755
--- a/cmd/snap-confine/tests/test_restrictions_working_args_termios
+++ b/cmd/snap-confine/tests/test_restrictions_working_args_termios
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 get_common_syscalls >"$TMP"/tmpl
diff --git a/cmd/snap-confine/tests/test_unrestricted b/cmd/snap-confine/tests/test_unrestricted
index dede157..d0f908a 100755
--- a/cmd/snap-confine/tests/test_unrestricted
+++ b/cmd/snap-confine/tests/test_unrestricted
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 cat >"$TMP/snap.name.app" <<EOF
diff --git a/cmd/snap-confine/tests/test_unrestricted_missed b/cmd/snap-confine/tests/test_unrestricted_missed
index d2473e9..a6396b9 100755
--- a/cmd/snap-confine/tests/test_unrestricted_missed
+++ b/cmd/snap-confine/tests/test_unrestricted_missed
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 cat >"$TMP/snap.name.app" <<EOF
diff --git a/cmd/snap-confine/tests/test_whitelist b/cmd/snap-confine/tests/test_whitelist
index 78c255d..b245f03 100755
--- a/cmd/snap-confine/tests/test_whitelist
+++ b/cmd/snap-confine/tests/test_whitelist
@@ -2,6 +2,7 @@
 
 set -e
 
+# shellcheck source=snap-confine/tests/common.sh
 . "${srcdir:-.}/snap-confine/tests/common.sh"
 
 printf "Test appname whitelist"
-- 
2.10.2

